File: growsari-common/src/main/java/com/growsari/application/common/dto/PageableRequestDTO.java
package com.growsari.application.common.dto;

import javax.validation.constraints.Min;

/**
 * @author alexander.ballester
 */
public class PageableRequestDTO {
    private static final String PAGE_PROP = "page";
    private static final String SIZE_PROP = "size";
    @Min(0)
    private int page;
    @Min(1)
    private int size;

    public PageableRequestDTO(){

    }

    public PageableRequestDTO(int page, int size) {
        this.page = page;
        this.size = size;
    }

    public int getPage() {
        return page;
    }

    public void setPage(int page) {
        this.page = page;
    }

    public int getSize() {
        return size;
    }

    public void setSize(int size) {
        this.size = size;
    }
}

File: growsari-client/src/app/entities/dto/pageable-request-dto.ts
export class PageableRequestDTO {
    page: number;
    size: number;
    // sortDirection: Direction;
    sortProperty: string;

    constructor()
    
    constructor(page?: number, size?: number){
        this.page = page;
        this.size = size;
    }

    public getPage() {
        return this.page;
    }

    public setPage(page : number) {
        this.page = page;
    }

    public getSize() {
        return this.size;
    }

    public setSize(size : number) {
        this.size = size;
    }

    public getSortProperty() {
        return this.sortProperty;
    }

    public setSortProperty(sortProperty : string) {
        this.sortProperty = sortProperty;
    }
}
